airflow/providers/sftp/hooks/sftp.py

```diff
@@ -76,8 +75,6 @@ def __init__(self, ftp_conn_id: str = 'sftp_default', *args, **kwargs) -> None:
             conn = self.get_connection(self.ssh_conn_id)
             if conn.extra is not None:
                 extra_options = conn.extra_dejson
-                if 'private_key_pass' in extra_options:
-                    self.private_key_pass = extra_options.get('private_key_pass')
```

Comment:
This makes this PR a breaking change -- could you please add a back-compatiblity path in (i.e. something that notices when this option is set but the new one is not, uses it and issues a deprecation warning)?

Reply:
:heavy_check_mark:  Done